atakama/packager.py

```diff
@@ -0,0 +1,246 @@
+        parser.add_argument("--crt", help="Openssl certificate file", required=True)
+        parser.add_argument(
+            "--openssl", help="Location of openssl binary", default=which("openssl")
+        )
+        parser.add_argument(
+            "--self-signed", help="Allow a self-signed cert", action="store_true"
+        )
+        args = parser.parse_args(argv)
+        if not args.src and not args.pkg:
+            raise ValueError("Nothing to do: must specify --src or --pkg")
+        return args
+
+    def run_setup(self) -> None:
+        """Given a src directory, run the setup.py command within it."""
+        wd = os.getcwd()
+        try:
+            os.chdir(os.path.dirname(self.src))
+            print("running: ", os.getcwd(), self.setup_path, file=sys.stderr)
+            dist = run_setup(self.setup_path, script_args=["bdist_wheel"])
+            for typ, _, path in getattr(dist, "dist_files"):
+                if typ == "bdist_wheel":
+                    self.pkg = os.path.abspath(path)
+                    return
+            assert False, "Expected package not created"  # pragma: no cover
+        finally:
+            os.chdir(wd)
+
+    def has_setup(self):
+        """True if the setup.py file exists."""
+        return os.path.exists(self.setup_path)
+
+    def make_setup(self):
+        """Make a fake setup.py.  This is probably a bad idea."""
+        print("make setup:", self.setup_path, file=sys.stderr)
+        self.made_setup = True
+        with open(self.setup_path, "w") as f:
+            f.write(
+                textwrap.dedent(
+                    """
+        from setuptools import setup
+        setup(
+            name="{plug_name}",
+            version="1.0",
+            description="{plug_name}",
+            packages=["{plug_name}"],
+            setup_requires=["wheel"],
+        )
+            """.format(
+                        plug_name=os.path.basename(self.src)
+                    )
+                )
+            )
+
+    def remove_setup(self):
+        os.remove(self.setup_path)
+
+    def openssl(self, cmd, **kws) -> subprocess.CompletedProcess:
+        """Shell openssl, and log that you did so."""
+        print("+ openssl", " ".join(cmd), file=sys.stderr)
+        cmd = [self.openssl_path] + cmd
+        return subprocess.run(cmd, **kws)  # pylint: disable=subprocess-run-check
+
+    def sign_package(self):
+        """Shell openssl, to sign a package given a crt and a privkey."""
+        key = self.key
+        crt = self.crt
+        pkg = self.pkg
+
+        sig = pkg + ".sig"
+        self.openssl(["dgst", "-sha256", "-sign", key, "-out", sig, pkg], check=True)
+        self.verify_signature(crt, pkg, sig)
+
+    @staticmethod
+    def verify_certificate(crt):
+        """Validate a cert against system root certs."""
+        with open(crt, "rb") as f:
+            end_entity_cert = f.read()
+        validator = CertificateValidator(end_entity_cert)
```

Comment:
AFAICT this checks CRLs, which is nice. Maybe a comment to explain why we don't just `openssl-verify`

Reply:
ok, will do